### PR TITLE
Fixed dimensions (width and height) of AbstractScrollArea (issue 9351)

### PR DESCRIPTION
**Pull request by mever**
As requested in: #9351

### CODE REVIEW

**Comment by mever** [comment]
Seems I need to fix some problems first.

**Comment by level420** [comment]
@mever could you please review your PR, as it seems that it makes some other tests fail. See https://travis-ci.org/qooxdoo/qooxdoo/jobs/250567447

Those are:
* qx.test.ui.core DragDropScrolling:testIsScrollbarVisible FAILED
* qx.test.ui.core DragDropScrolling:testIsScrollbarExceedingMaxPos FAILED
* qx.test.ui.core DragDropScrolling:testScrollBy FAILED

Maybe those tests rely on the pre defined minimum scroll area dimensions.

File: framework/source/class/qx/ui/core/scroll/AbstractScrollArea.js [hunk 1]
```diff
@@ -267,9 +257,10 @@ qx.Class.define("qx.ui.core.scroll.AbstractScrollArea",
     */
 
     /**
-     * Returns the boundaries of the pane.
+     * Returns the dimensions of the pane.
      *
-     * @return {Map} The pane boundaries.
+     * @return {{width: number, height: number}|null} The pane dimension in pixel. Contains
```
Comment (derrell) [inline framework/source/class/qx/ui/core/scroll/AbstractScrollArea.js]: This is a useful depiction when read in the code, but what's in the curly brackets is supposed to be a type. Does this render properly in the API viewer? What would typically be there is:
```
@return {Map|null} The pane dimensions in pixels. The Map contains two members: width and height, the respective number of pixels for that dimension.
```

I like what you've shown, but I don't think we should put it in one place but not globally. It would be a nice global change.

I'm open to accepting this change as is, if others feel it's appropriate.

**Review by derrell — CHANGES_REQUESTED** [review]

File: framework/source/class/qx/ui/core/scroll/AbstractScrollArea.js (hunk 1 above)
Comment (johnspackman) [inline framework/source/class/qx/ui/core/scroll/AbstractScrollArea.js]: i'll go with this unless it breaks the API viewer or jsdoc linting - I've often wished for a decent way to describe map members

**Review by johnspackman — APPROVED** [review]
approving, subject to consensus on the jsdoc

**Review by level420 — CHANGES_REQUESTED** [review]
@mever please have a look into the failing travis tests. Thank you.

**Review by cboulanger — APPROVED** [review]

File: framework/source/class/qx/ui/core/scroll/AbstractScrollArea.js (hunk 1 above)
Comment (mever) [inline framework/source/class/qx/ui/core/scroll/AbstractScrollArea.js]: This works well with my code editor (PhpStorm and by extension WebStorm from Jetbrains) which has Qooxdoo support build in. I think it uses JSDoc (http://usejsdoc.org/) to interpret the doc-blocks.

File: framework/source/class/qx/ui/core/scroll/AbstractScrollArea.js [hunk 2]
```diff
@@ -75,6 +75,12 @@ qx.Class.define("qx.ui.core.scroll.AbstractScrollArea",
       this._setLayout(grid);
     }
 
+    // since the scroll container disregards the min size of the scrollbars
+    // we have to set the min size of the scroll area to ensure that the
+    // scrollbars always have an usable size.
+    var size = qx.ui.core.scroll.AbstractScrollArea.DEFAULT_SCROLLBAR_WIDTH;
+    this.set({minHeight: size, minWidth: size});
+
```
Comment (level420) [inline framework/source/class/qx/ui/core/scroll/AbstractScrollArea.js]: Following the logic in the comment above, the use of this is to initialize scroll area with minimal dimensions. It seems that we need place for both a horizontal and a vertical scrollbar. This is minimum `2 * qx.ui.core.scroll.AbstractScrollArea.DEFAULT_SCROLLBAR_WIDTH`.

But I doubt that this is enough as the text states that we should have a USABLE size which shows the up/down or left/right button of the scrollbar and maybe a little room for the scroll knob itself.

With having `qx.ui.core.scroll.AbstractScrollArea.DEFAULT_SCROLLBAR_WIDTH` set to 14, the former minWidth of 50 is not so bad (14 *2 + 22 pixel to scroll).

And a second point is that this change introduces `minHeight` and `minWidth` for all themes deriving from `qx.theme.simple.Appearance` which did not define `minHeight` and `minWidht` through the `scrollarea` appearance.

**Review by level420 — CHANGES_REQUESTED** [review]

File: framework/source/class/qx/ui/core/scroll/AbstractScrollArea.js (hunk 1 above)
Comment (level420) [inline framework/source/class/qx/ui/core/scroll/AbstractScrollArea.js]: @mever did you build the api viewer with your PR? I not: please do so and see if the api viewer shows this as expected, containing your changes. It it is OK leave it in. If not, please revert the change. Thank you.

**Comment by level420** [comment]
I've done some investigations why some tests in `qx.test.ui.core.DragDropScrolling` fail. It is because the list element has no default height set. If we define a height for the list element like this:
```javascript
    _createDummyList : function()
    {
      var list = new qx.ui.form.List();
      list.setHeight(50); // <--- SET HEIGHT HERE
      for (var i=0; i<20; i++) {
        list.add(new qx.ui.form.ListItem("Item " + i));
      }
      return list;
    },
```
Tests are successful again.
For some reason it seems that `qx.ui.core.scroll.AbstractScrollArea` needs a height set in order to calculate visibility of scrollbars.

@mever could you please have a look into the failing tests? You can see them in the travis runs.

**Comment by level420** [comment]
I've run this PR with one of my apps without any problems, but I don't use any `qx.ui.form.List` widgets there.

**Comment by mever** [comment]
I've added some commits that pass the DragDropScrolling tests locally.

**Review by level420 — APPROVED** [review]
We previously had a default width and height in the abstract scroll area of 100/200 which is gone and have it now in th list. Seems OK!

File: framework/source/class/qx/ui/core/scroll/AbstractScrollArea.js (hunk 1 above)
Comment (mever) [inline framework/source/class/qx/ui/core/scroll/AbstractScrollArea.js]: I just build with the API viewer. The latest changes also show the width and height nicely in the documentation page of the form.List. No problems in the API viewer anymore.

**Review by derrell — APPROVED** [review]

**Comment by level420** [comment]
4 approvals, more than 2 business days, merging, thank you @mever
